core/object/undo_redo.cpp

```diff
@@ -80,14 +80,14 @@ bool UndoRedo::_redo(bool p_execute) {
 	return true;
 }
 
-void UndoRedo::create_action(const String &p_name, MergeMode p_mode) {
+void UndoRedo::create_action(const String &p_name, MergeMode p_mode, bool p_backward_undo_ops) {
 	uint64_t ticks = OS::get_singleton()->get_ticks_msec();
 
 	if (action_level == 0) {
 		_discard_redo();
 
 		// Check if the merge operation is valid
-		if (p_mode != MERGE_DISABLE && actions.size() && actions[actions.size() - 1].name == p_name && actions[actions.size() - 1].last_tick + 800 > ticks) {
+		if (p_mode != MERGE_DISABLE && actions.size() && actions[actions.size() - 1].name == p_name && actions[actions.size() - 1].backward_undo_ops == p_backward_undo_ops && actions[actions.size() - 1].last_tick + 800 > ticks) {
```

Comment:
This change is not really necessary. It doesn't make sense to make the same action both reverse and not reverse, so the new condition will always be true (unless someone does something weird).

Reply:
then should I change it to an error or should i explicit in documentation that `p_backward_undo_ops` should be consistent for merging. I feel like without it people might not understand what wrong is happening when they try to merge it and their order is all messed up.

Reply:
Hmm :/
Maybe it's better to leave it.